capstone/src/main/java/com/google/sps/data/Room.java

```diff
@@ -0,0 +1,98 @@
+package com.google.sps.data;
+
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.Entity;
+import java.util.ArrayList;
+import java.util.List;
+import com.google.sps.service.DatabaseService;
+
+public class Room {
+    public static final String ROOM_ENTITY_NAME = "Room";
+    public static final String TITLE_PROPERTY_KEY = "title";
+    public static final String DESCRIPTION_PROPERTY_KEY = "description";
+    public static final String HOST_PROPERTY_KEY = "host";
+    public static final String FOLLOWERS_PROPERTY_KEY = "followers";
+    public static final String FORMS_PROPERTY_KEY = "forms";
+    
+
+    private Entity entity;
+
+    public Room(Entity entity) {
+        this.entity = entity;
+    }
+
+    public Room(User host, String title, String description) { 
+        this.entity = new Entity(Room.ROOM_ENTITY_NAME);
+        this.entity.setProperty(Room.TITLE_PROPERTY_KEY, title);
+        this.entity.setProperty(Room.DESCRIPTION_PROPERTY_KEY, description);
+        this.entity.setProperty(Room.HOST_PROPERTY_KEY, host.getUserKey());
+        this.entity.setProperty(Room.FOLLOWERS_PROPERTY_KEY, null);
+        this.entity.setProperty(Room.FORMS_PROPERTY_KEY, null);
+    }
+
+    public Entity getRoomEntity() {
+        return this.entity;
+    }
+
+    public Key getHost() {
+        return (Key) this.entity.getProperty(Room.HOST_PROPERTY_KEY);
+    }
+
+    public void setHost(User host) { 
+        this.entity.setProperty(Room.HOST_PROPERTY_KEY, host.getUserKey());
+    }
+
+    public String getTitle() {
+        return (String) this.entity.getProperty(Room.TITLE_PROPERTY_KEY);
+    }
+
+    public void setTitle(String title){
+        this.entity.setProperty(Room.TITLE_PROPERTY_KEY, title);
+    }
+
+    @SuppressWarnings("unchecked")
+    public List<Key> getAllFollowers() {
+        return (ArrayList<Key>) this.entity.getProperty(Room.FOLLOWERS_PROPERTY_KEY);
+    }
+
+    @SuppressWarnings("unchecked")
+    public List<Key> getAllForms() { 
+        return (ArrayList<Key>) this.entity.getProperty(Room.FORMS_PROPERTY_KEY); 
+    }
+    
+    @SuppressWarnings("unchecked")
+    public void addFollower(User follower) {
+        List<Key> followers = (ArrayList<Key>) this.entity.getProperty(Room.FOLLOWERS_PROPERTY_KEY);
+        if(followers == null){
+            followers = new ArrayList<>();
+        }
+        followers.add(follower.getUserKey());
```

Comment:
In the case that it was null, it sets followers to a new list, but that new list is not actually on the entity, so the new list with one item in it just gets dropped on the floor after this function exits. Add a test to cover this case (adding the first follower to a Room created with a preexisting entity that does not have followers)

Reply:
Got it. Thanks to @kyleasmith1  for explaining it more clearly to me.